conda/models/match_spec.py

```diff
@@ -222,6 +222,11 @@ def is_name_only_spec(self):
                 and 'name' in self._match_components
                 and self.name != '*')
 
+    @property
+    def is_prerelease(self) -> bool:
+        """True if spec contains an exact version spec for a prerelease version."""
+        return self.version is not None and self.version.is_prerelease
```

Comment:
Note that version specs can have `*` globs, so I don't know how the pre-release regex string is handling those, or what the answer should be. I'd like to hear your thoughts on this... I guess it should depend on the context option chosen?